src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php

```diff
@@ -418,20 +414,45 @@ protected function executeStatement(string $sql, array $parameters = [], array $
         try {
             $stmt = $this->driverConnection->executeStatement($sql, $parameters, $types);
         } catch (TableNotFoundException $e) {
-            if ($this->driverConnection->isTransactionActive()) {
+            if (!$this->autoSetup || $this->driverConnection->isTransactionActive()) {
                 throw $e;
             }
 
-            // create table
-            if ($this->autoSetup) {
-                $this->setup();
-            }
+            $this->setup();
+
             $stmt = $this->driverConnection->executeStatement($sql, $parameters, $types);
         }
 
         return $stmt;
     }
 
+    private function executeInsert(string $sql, array $parameters = [], array $types = []): string
+    {
+        $this->driverConnection->beginTransaction();
+
+        try {
+            try {
+                $this->driverConnection->executeStatement($sql, $parameters, $types);
+            } catch (TableNotFoundException $e) {
+                if (!$this->autoSetup) {
+                    throw $e;
+                }
+
+                $this->setup();
+
+                $this->driverConnection->executeStatement($sql, $parameters, $types);
+            }
+
+            if (!$id = $this->driverConnection->lastInsertId()) {
+                throw new TransportException('lastInsertId() returned false, no id was returned.');
+            }
+        } finally {
+            $this->driverConnection->commit();
+        }
+
+        return $id;
```

Comment:
@nicolas-grekas I removed the (string) casting, but won't this cause a type error without it? lastInsertId() returns string, int or false.

Reply:
Nevermind, I added a test for it and we're good!

Reply:
Except psalm doesn't like it. I need the (string) to satisfy psalm.